python-package/xgboost/spark/core.py

```diff
@@ -527,17 +547,69 @@ def _fit(self, dataset):
 
         select_cols = [label_col]
         features_cols_names = None
-        if self.getOrDefault(self.features_cols):
-            features_cols_names = self.getOrDefault(self.features_cols)
-            features_cols = _validate_and_convert_feature_col_as_float_col_list(
-                dataset, features_cols_names
-            )
-            select_cols.extend(features_cols)
+        enable_sparse_data_optim = self.getOrDefault(self.enable_sparse_data_optim)
+        if enable_sparse_data_optim:
+            from pyspark.ml.linalg import VectorUDT
+
+            if self.getOrDefault(self.missing) != 0.0:
```

Comment:
could we put this checking into the _validate_params?

Reply:
@trivialfis 
According to your comment: https://github.com/dmlc/xgboost/pull/8145#discussion_r943352000
Seemingly we don't need to add this restriction ?

Reply:
I think we need the restriction that `missing` must be 0. Otherwise there will be two missing/invalid value, 0s removed by spark. `missing` removed by xgboost.

Reply:
@trivialfis 

OK. So that CSR input is still different with dense input when handling "missing",
For CSR input, both inactive elements (removed elements in CSR) or elements of `value==${missing param value}` are regarded as missing value.

We should document this carefully. Dense / CSC / CSR are different on missing logic.